pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -663,87 +663,23 @@ func (m *cgroupManagerImpl) GetCgroupConfig(name CgroupName, resource v1.Resourc
 	return nil, fmt.Errorf("unsupported resource %v for cgroup %v", resource, name)
 }
 
-func setCgroupv1CpuConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	var cpuQuotaStr, cpuPeriodStr, cpuSharesStr string
-	if resourceConfig.CPUQuota != nil {
-		cpuQuotaStr = strconv.FormatInt(*resourceConfig.CPUQuota, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.cfs_quota_us"), []byte(cpuQuotaStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuQuotaStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUPeriod != nil {
-		cpuPeriodStr = strconv.FormatUint(*resourceConfig.CPUPeriod, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.cfs_period_us"), []byte(cpuPeriodStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuPeriodStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUShares != nil {
-		cpuSharesStr = strconv.FormatUint(*resourceConfig.CPUShares, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.shares"), []byte(cpuSharesStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuSharesStr, cgroupPath, err)
-		}
-	}
-	return nil
-}
-
-func setCgroupv2CpuConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	if resourceConfig.CPUQuota != nil {
-		if resourceConfig.CPUPeriod == nil {
-			return fmt.Errorf("CpuPeriod must be specified in order to set CpuLimit")
-		}
-		cpuLimitStr := Cgroup2MaxCpuLimit
-		if *resourceConfig.CPUQuota > -1 {
-			cpuLimitStr = strconv.FormatInt(*resourceConfig.CPUQuota, 10)
-		}
-		cpuPeriodStr := strconv.FormatUint(*resourceConfig.CPUPeriod, 10)
-		cpuMaxStr := fmt.Sprintf("%s %s", cpuLimitStr, cpuPeriodStr)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.max"), []byte(cpuMaxStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuMaxStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUShares != nil {
-		cpuWeight := CpuSharesToCpuWeight(*resourceConfig.CPUShares)
-		cpuWeightStr := strconv.FormatUint(cpuWeight, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.weight"), []byte(cpuWeightStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuWeightStr, cgroupPath, err)
-		}
+// Set resource config for the specified resource type on the cgroup
+func (m *cgroupManagerImpl) SetCgroupConfig(name CgroupName, resourceConfig *ResourceConfig) error {
+	containerConfig := &CgroupConfig{
+		Name:               name,
+		ResourceParameters: resourceConfig,
 	}
-	return nil
-}
 
-func setCgroupCpuConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	if libcontainercgroups.IsCgroup2UnifiedMode() {
-		return setCgroupv2CpuConfig(cgroupPath, resourceConfig)
-	} else {
-		return setCgroupv1CpuConfig(cgroupPath, resourceConfig)
+	libcontainerCgroupConfig := m.libctCgroupConfig(containerConfig, true)
+	manager, err := libcontainercgroupmanager.New(libcontainerCgroupConfig)
+	if err != nil {
+		return err
```

Comment:
are we sure the error will contain the cgroup name and the resource we tried to set? Perhaps wrap the error for easier debugging to be on a safe side. Hopefully this will never happen so extra wrap is not affecting anything

Reply:
Thanks, done!